CMS-Enterprise/mint-app
# [NOREF] Update TruncatedText component to take line clamp prop

# [NOREF]

## Description

The `TruncatedText` component currently only takes a character limit. But recently, there have been designs asking to truncate text to _x_
 amount of lines (line clamping) instead of character count. This PR updates the `TruncatedText` component to take in line clamp value

This PR proposes moving away from using `MentionTextArea` to parse and render the text with all of its text entity and uses [html-react-parser](https://github.com/remarkablemark/html-react-parser) to parse and render said text

## How to test this change

1. `yarn install` && restart servers
2.  Navigate to a CR and TDL side panel and see that CR Summary is line clamped to 5, instead of a character count.
3. Look elsewhere in the app to ensure any other instances of `TruncatedText` still renders correct

## PR Author Checklist




## PR Reviewer Guidelines

- It's best to pull the branch locally and test it, rather than just looking at the code online!
- When approving a PR, provide a reason _why_ you're approving it
  - e.g. "Approving because I tested it locally and all functionality works as expected"
  - e.g. "Approving because the change is simple and matches the Figma design"
- Don't be afraid to leave comments or ask questions, especially if you don't understand why something was done! (This is often a great time to suggest code comments or documentation updates)
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
